pennylane/devices/default_mixed.py

```diff
@@ -152,7 +155,7 @@ def state(self):
         """Returns the state density matrix of the circuit prior to measurement"""
         dim = 2**self.num_wires
         # User obtains state as a matrix
-        return self._reshape(self._pre_rotated_state, (dim, dim))
+        return self._reshape(self._state, (dim, dim))
```

Comment:
Right, this was just a quick way to get the format that I wanted. Basically `self._state` is not a `2^n x 2^n` matrix, but some other tensor shape. `self.state` and `self.density_matrix` on the other hand return the matrix that one expects. However, since `self.state` actually returns `self._pre_rotated_state`, so changes to the quantum state are not reflected in this property until [all operations have been applied](https://github.com/PennyLaneAI/pennylane/blob/99dd676bf0647e0859614aa996c9da72332b3087/pennylane/devices/default_mixed.py#L493-L497), and the same goes for `self.density_matrix` since it just calls `self.state`.